pilot/pkg/serviceregistry/kube/controller/namespacecontroller.go

```diff
@@ -72,10 +80,22 @@ func NewNamespaceController(kubeClient kube.Client, caBundleWatcher *keycertbund
 			// skip special kubernetes system namespaces
 			return false
 		}
+		if features.EnableEnhancedResourceScoping && !c.DiscoveryNamespacesFilter.Filter(o) {
```

Comment:
nit: I would have the namespacecontroller not know about EnableEnhancedResourceScoping; instead just conditionally pass DiscoveryNamespacesFilter based on EnableEnhancedResourceScoping, and make this a nil-check.

Just a minor style thing to move the feature flags up into higher level code instead of in the core.

Reply:
Done